src/Components/test/E2ETest/e2eTestSettings.ci.json

```diff
@@ -0,0 +1,10 @@
+{
+  // We give each Selenium test assertion up to two minutes to fail before any other test in the
+  // build has failed, after that, we fail after 5 seconds.
+  "DefaultWaitTimeoutInSeconds": 120,
+  // Worst case scenario for 1000 failing tests will cause the build to fail after 84 minutes.
+  // We currently have < 500 E2E tests so even if all tests fail it will take around 40 minutes to do so.
```

Comment:
This sort of comment is guaranteed to become out of date. I don't think you have to give an explanation of how long the whole tests run will take. The following comment:

> This value is balanced between completing the build fast enough upon failure and giving
> each E2E test a fair chance to pass even in the event that a separate test has failed already.

... is already enough of a justification IMO.

Reply:
That's fair, maybe the comment belonged on the PR. I wanted to make clear that I did the math to pick the values.